redbot/cogs/audio/apis/interface.py

```diff
@@ -299,74 +229,51 @@ async def _spotify_first_time_query(
                 if youtube_cache and val:
                     task = ("update", ("youtube", {"track": track_info}))
                     self.append_task(ctx, *task)
-
                 if val:
                     youtube_urls.append(val)
             else:
                 youtube_urls.append(track_info)
-            await asyncio.sleep(0)
             track_count += 1
-            if notifier and ((track_count % 2 == 0) or (track_count == total_tracks)):
+            if notifier is not None and ((track_count % 2 == 0) or (track_count == total_tracks)):
                 await notifier.notify_user(current=track_count, total=total_tracks, key="youtube")
         if CacheLevel.set_spotify().is_subset(current_cache_level):
             task = ("insert", ("spotify", database_entries))
             self.append_task(ctx, *task)
         return youtube_urls
 
-    async def _youtube_first_time_query(
-        self,
-        ctx: commands.Context,
-        track_info: str,
-        current_cache_level: CacheLevel = CacheLevel.none(),
-    ) -> str:
-        track_url = await self.youtube_api.get_call(track_info)
-        if CacheLevel.set_youtube().is_subset(current_cache_level) and track_url:
-            time_now = int(datetime.datetime.now(datetime.timezone.utc).timestamp())
-            task = (
-                "insert",
-                (
-                    "youtube",
-                    [
-                        {
-                            "track_info": track_info,
-                            "track_url": track_url,
-                            "last_updated": time_now,
-                            "last_fetched": time_now,
-                        }
-                    ],
-                ),
-            )
-            self.append_task(ctx, *task)
-        return track_url
-
-    async def _spotify_fetch_tracks(
+    async def fetch_from_spotify_api(
         self,
         query_type: str,
         uri: str,
         recursive: Union[str, bool] = False,
         params: MutableMapping = None,
         notifier: Optional[Notifier] = None,
-    ) -> Union[MutableMapping, List[str]]:
+        ctx: Context = None,
+    ) -> Union[List[MutableMapping], List[str]]:
+        """Gets track info from spotify API"""
 
         if recursive is False:
-            (call, params) = self._spotify_format_call(query_type, uri)
-            results = await self.spotify_api.get_call(call, params)
+            (call, params) = self.spotify_api.spotify_format_call(query_type, uri)
+            results = await self.spotify_api.make_get_call(call, params)
         else:
-            results = await self.spotify_api.get_call(recursive, params)
+            if isinstance(recursive, str):
+                results = await self.spotify_api.make_get_call(recursive, params)
+            else:
+                results = {}
         try:
             if results["error"]["status"] == 401 and not recursive:
                 raise SpotifyFetchError(
```

Comment:
THIS IS PART 3/8 THAT NEED TO BE APPLIED TOGETHER WITH THIS.

*oof, I can't make a suggestion cause it's not part of the changes, GitHub why...*

You need to replace `(` with `_(` line below the one I commented on